Bootstrap login for self-service account pages documentation (#2116)
FusionAuth/fusionauth-site: site/docs/v1/tech/account-management/bootstrapping-login.adoc

---
layout: doc

> title: Bootstrapping Login For Self Service Account Management

Reviewer: ```suggestion
title: Bootstrapping Login For Self-Service Account Management
```

description: Allow users to access self-service account management from external applications
---

include::docs/v1/tech/shared/_premium-edition-blurb.adoc[]

[NOTE.since]
====
Available since 1.45.0
====

> == Accessing the Self Service Account Pages From External Applications

Reviewer: ```suggestion
== Accessing the Self-Service Account Pages From External Applications
```

> In some scenarios you may want to be able to allow users to access the self-service account management pages when it is impractical to use a normal login flow. One such scenario is with native mobile applications where the user is logged in to FusionAuth and the application is in possession of the access token. The following procedure will allow you to access these pages without requiring the user to log in again.

Reviewer: ```suggestion
In some scenarios you may want to allow users to access the self-service account management pages when it is impractical to use a normal login flow. One such scenario is with native mobile applications where the user is logged in to FusionAuth and the application is in possession of the access token. The following procedure will allow you to access these pages without requiring the user to log in again.
```

> === Opening the Self Account Pages With an Access Token

Reviewer: ```suggestion
=== Opening the Self-Service Account Pages With an Access Token
```

In order to access the self-service account management pages in this manner you will need to meet the following criteria:

* The user is logged in
* The application is in possession of the access token JWT

> * The application can open a browser window or webview and include an Authorization header

Reviewer: Is webview a generic term? Or should it be WebView?

Author: I think WebView is technically an Android thing but iOS people will know (it's like UIView in native iOS and otherwise dependant on what framework you chose to build both)

> 

Author: it does

> If those criteria are met you can have the application open the web view or browser window with a `GET` request to the account self-service page and include the access token JWT as a Bearer token in the Authorization header. You must also include the client Id of the application.

Reviewer: ```suggestion
If the above criteria are met, you can open a webview or browser window from the external application and make a `GET` request to the account self-service page that includes the access token JWT as a Bearer token in the Authorization header. You must also include the client Id of the application.
```

[source,shell]
.Example HTTP Request with header
GET /account/?client_id=<your_client_id> HTTP/1.1
Authorization: Bearer <your_access_token>

[NOTE]
====

> Because you are only passing in an access token and the web view has no session state the interactions with the self-service account management pages will only succeed as long as the access token is valid and not expired. You may need to refresh the token prior to opening the self-service account management page or adjust the JWT lifespan to accommodate.

Reviewer: ```suggestion
Because you are only passing in an access token and the webview has no session state, the interactions with the self-service account management pages will only succeed as long as the access token is valid and not expired. You may need to refresh the token prior to opening the self-service account management page or adjust the JWT lifespan to accommodate.
```
